### DISCUSSION TITLE
Where does it store the file with the secret key?

### DISCUSSION THREAD

**Discussion by vporton** [oxen-io/session-desktop#2168] [Q&A] [answered]
Where does it store the file with the secret key (on Linux)?

**Comment by Bilb** [comment, marked as answer]
Attachments are stored in `~/.config/Session/attachments.noindex`.
The key used to encrypt/decrypt attachments is stored in the database itself which is encrypted too with your app password or a generated one if you don't have any set.

In the sqlite database, the attachment encrypting key is stored in the key `local_attachment_encrypted_key` of the table 'items'
This is a generated key, not user entered. Basically on start, if none are set we generate one (in `data.ts` `generateAttachmentKeyIfEmpty()`

That key is then used for encrypting attachments on save and decrypting them on read.
Is that all you need?
